lib/Jsrt/JsrtHelper.cpp

```diff
@@ -89,40 +91,32 @@ void JsrtCallbackState::ObjectBeforeCallectCallbackWrapper(JsObjectBeforeCollect
     }
 
 #ifndef _WIN32
-    static pthread_key_t s_threadLocalDummy;
+    static THREAD_LOCAL struct _OnThreadExit
+    {
+        ~_OnThreadExit()
+        {
+            ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread();
+        }
+    } _s_onThreadExit;
 #endif
-    static THREAD_LOCAL bool s_threadWasEntered = false;
 
-    _NOINLINE void DISPOSE_CHAKRA_CORE_THREAD(void *_)
```

Comment:
I don't know as it is hard to repro/investigate. (Some say pthread_key_create destructors might be skipped if pthread_exit is not called, e.g. by calling exit -- not sure if that plays a role).